ios/brave-ios/Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/Scripts/Sandboxed/SelectorsPollerScript.js

```diff
@@ -1070,19 +1084,96 @@ window.__firefox__.execute(function($) {
       unhideSelectorsMatchingElementsAndTheirParents(nodesWithExtractedURLs)
     }
 
-    // 4. Set the rules on the stylesheet. So far we don't have any rules set
+    // 4. Perform procedural actions before setting stylesheet
+    if (CC.hasProceduralActions) {
+      executeProceduralActions()
+    }
+
+    // 5. Set the rules on the stylesheet. So far we don't have any rules set
     // We could do this earlier but it causes elements to hide and unhide
     // It's best to wait to this period so we have
     setRulesOnStylesheet()
-    // 5. Start our queue pump if we need to
+    // 6. Start our queue pump if we need to
     scheduleQueuePump(args.hideFirstPartyContent, args.genericHide)
 
     if (!args.hideFirstPartyContent) {
-      // 6. Start listening to new urls
+      // 7. Start listening to new urls
       startURLMutationObserver()
     }
   }
 
+  /// Process any :style, :remove, :remove-attr, or :remove-class selectors
+  const executeProceduralActions = (added) => {
+    // If passed a list of added elements, do not query the entire document
+    if (CC.proceduralActionFilters === undefined) {
+      return
+    }
+
+    /**
+    * Adds given styles to allStyleRules
+    * @param {Object} entry The entry to add
+    */
+    const addStyleRule = (entry) => {
+      const rule = entry.selector + '{' + entry.rules.join(';') + ';}'
```

Comment:
`entry` is always 1 element long; no need to accept an array here

Reply:
i don't know if I updated the code (as this used to be originally an array in the past) but entry isn't an array here so im not sure what if anything I'm supposed to do here.